From 40f0e162e7556c604a60bb04946ba9ff142c4f1e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Beh=C3=BAn?= <kabel@kernel.org>
Date: Tue, 5 Mar 2024 11:06:13 +0100
Subject: [PATCH] mvebu: patches-5.15: Change MCU compatible to
 cznic,turris-omnia-ctrl
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is a temporary solution so that U-Boot does not inject reset-gpios
into PCIe nodes, because it breaks some WiFi cards.

Signed-off-by: Marek Behún <kabel@kernel.org>
---
 ...urris-omnia-mcu-Change-compatible-to.patch | 35 +++++++++++++++++++
 1 file changed, 35 insertions(+)
 create mode 100644 target/linux/mvebu/patches-5.15/990-platform-cznic-turris-omnia-mcu-Change-compatible-to.patch

diff --git a/target/linux/mvebu/patches-5.15/990-platform-cznic-turris-omnia-mcu-Change-compatible-to.patch b/target/linux/mvebu/patches-5.15/990-platform-cznic-turris-omnia-mcu-Change-compatible-to.patch
new file mode 100644
index 0000000000..bdb98d7787
--- /dev/null
+++ b/target/linux/mvebu/patches-5.15/990-platform-cznic-turris-omnia-mcu-Change-compatible-to.patch
@@ -0,0 +1,35 @@
+From: =?UTF-8?q?Marek=20Beh=C3=BAn?= <kabel@kernel.org>
+Date: Tue, 5 Mar 2024 10:42:07 +0100
+Subject: platform: cznic: turris-omnia-mcu: Change compatible to
+ cznic,turris-omnia-ctrl
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+This is a temporary solution so that U-Boot does not inject reset-gpios
+into PCIe nodes, because it breaks some WiFi cards.
+
+Signed-off-by: Marek Behún <kabel@kernel.org>
+
+--- a/arch/arm/boot/dts/armada-385-turris-omnia.dtsi
++++ b/arch/arm/boot/dts/armada-385-turris-omnia.dtsi
+@@ -201,7 +201,7 @@
+ 			reg = <0>;
+ 
+ 			mcu: system-controller@2a {
+-				compatible = "cznic,turris-omnia-mcu";
++				compatible = "cznic,turris-omnia-ctrl";
+ 				reg = <0x2a>;
+ 
+ 				pinctrl-names = "default";
+--- a/drivers/platform/cznic/turris-omnia-mcu-base.c
++++ b/drivers/platform/cznic/turris-omnia-mcu-base.c
+@@ -373,7 +373,7 @@ static int omnia_mcu_probe(struct i2c_cl
+ }
+ 
+ static const struct of_device_id of_omnia_mcu_match[] = {
+-	{ .compatible = "cznic,turris-omnia-mcu" },
++	{ .compatible = "cznic,turris-omnia-ctrl" },
+ 	{}
+ };
+ 
-- 
2.43.0

